src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java

```diff
@@ -252,9 +257,60 @@ private Panel getImgViewer() {
         Column<ImageInfo, String> imageTpsColumn = imageInfoGrid.addColumn(ImageInfo::getTimePoints).setCaption("Image Time Points");
         Column<ImageInfo, String> imageChannelsColumn = imageInfoGrid.addColumn(ImageInfo::getChannels).setCaption("Channels");
         Column<ImageInfo, Component> imageFullColumn = imageInfoGrid.addColumn(imageInfo -> (Component) linkToFullImage(imageInfo.getImageId())).setCaption("Full Image");
+        Column<ImageInfo, Button> imageMetadataColumn = imageInfoGrid.addColumn(imageInfo -> {
+            Button metadataButton = new Button("Show Metadata");
+            metadataButton.addClickListener(clickEvent -> {
+                Window metadataSubWindow = new Window("Metadata Sub-Window");
+                VerticalLayout metadataLayout = new VerticalLayout();
+
+                //TODO get metadata and add components
+                Collection<MetadataProperty> metadataProperties = new ArrayList<>();
+                long currentImageId = imageInfo.getImageId();
+                try {
+                    List metadataList = omeroClient.fetchMapAnnotationDataForImage(currentImageId);
+                    for (int i = 0; i < metadataList.size(); i++) {
+                        MapAnnotationData currentMapAnnotation = (MapAnnotationData) metadataList.get(i);
+                        List<NamedValue> list = (List<NamedValue>) currentMapAnnotation
+                            .getContent();
+                        for (NamedValue namedValue : list) {
+                            String metaDataKey = namedValue.name;
+                            String metaDataValue = namedValue.value;
+                            metadataProperties.add(new MetadataProperty<String>("Key: " + metaDataKey, "Value " + metaDataValue, "example property no."));
```

Comment:
Where do I find this in the portlet? Is it displayed after I clicked the button? I would expect this to be shown somewhere, or am I wrong?

Reply:
yes, this is inside a click listener to the metadata button.

Reply:
Nice Idea! Moved the setup into seperate method in b69b831